osu.Game/Tests/Visual/Spectator/TestSpectatorClient.cs

```diff
@@ -27,12 +28,22 @@ public class TestSpectatorClient : SpectatorClient
 
         public override IBindable<bool> IsConnected { get; } = new Bindable<bool>(true);
 
+        public readonly Dictionary<int, ReplayFrame> LastReceivedFrame = new Dictionary<int, ReplayFrame>();
```

Comment:
Is this public just for the test? If it is, then how about using `internal` here?

Also the name doesn't sit 100% well with me given it's talking about a singular "last received frame" while (I think) it's a dict storing last received frames for each user individually. The `spectatorClient.LastReceivedFrame.First()` call in the test scene was kind of doing my head in (because why call `.First()` on what appears to be a singular frame naming-wise). Maybe `LastReceivedUserFrames`?

Reply:
Agree with `LastReceivedUserFrames`. Also should probably limit exposure to `IReadOnlyDictionary` if we can.

Reply:
Have made this change